cluster-autoscaler/estimator/estimator.go

```diff
@@ -39,7 +39,7 @@ var AvailableEstimators = []string{BinpackingEstimatorName}
 // It returns the number of new nodes needed as well as the list of pods it managed
 // to schedule on those nodes.
 type Estimator interface {
-	Estimate([]*apiv1.Pod, *schedulerframework.NodeInfo, cloudprovider.NodeGroup) (int, []*apiv1.Pod)
+	Estimate([]*apiv1.Pod, *schedulerframework.NodeInfo, cloudprovider.NodeGroup, []BinpackingLimit) (int, []*apiv1.Pod)
```

Comment:
I'm not 100% sold on the runtime limits. I understand that this will make some implementation inside the orchestrator easier, but couldn't we have all the limits defined at the initialisation? We can give them a way to access some necessary variables either way.

Reply:
Ok, I am more in favour now. Otherwise we would need to pass many different arguments to the StartEstimation if we wanted to take into account values calculated during scale-up. Runtime limits SGTM then.

Reply:
I'm keeping them then :-)